runtime/lua/vim/fs.lua

```diff
@@ -86,6 +86,66 @@ function M.basename(file)
   return file:match('/$') and '' or (file:match('[^/]*$'))
 end
 
+--- @class vim.fs.abspath.Opts
+--- @inlinedoc
+---
+--- Current working directory to use as the base for the relative path.
+--- This option is ignored for C:foo\bar style paths on Windows, as those paths use the current
+--- directory of the drive specified in the path.
+--- (default: |current-directory|)
+--- @field cwd? string
+
+--- Convert path to an absolute path. Does not check if the path exists, resolve symlinks or
+--- hardlinks (including `.` and `..`), or expand environment variables. If the path is already
+--- absolute, it is returned unchanged.
+---
+--- @param path string Path
+--- @param opts vim.fs.abspath.Opts? Optional keyword arguments:
+--- @return string Absolute path
+function M.abspath(path, opts)
+  opts = opts or {}
+
+  vim.validate({
+    path = { path, { 'string' } },
+    cwd = { opts.cwd, { 'string' }, true },
+  })
+
+  local is_absolute --- @type boolean
+
+  if not iswin then
+    -- In Unix-like systems, an absolute path always starts with a slash.
+    is_absolute = path:sub(1, 1) == '/'
+  else
+    -- Windows absolute path formats:
+    --   {drive_letter}:\path\to\file: Absolute path from the root of drive.
+    --   \\{server}\{share}\path\to\file: UNC path.
+    --   \\{? or .}\{drive_letter}:\path\to\file: Device path.
+    --   \\{? or .}\UNC\{server}\{share}\path\to\file: Device UNC path.
+    --   \\{? or .}\Volume{GUID}\path\to\file: Device Volume path.
+    is_absolute = path:match('^%w:/')
+      or path:match('^//')
+      or path:match('^%w:\\')
+      or path:match('^\\\\')
+  end
```

Comment:
could be a ternary, which also avoids the ned for a `@type` annotation

Reply:
Actually, this entire code can be removed once #28203 is merged, since that allows us a much simpler way of detecting absolute paths